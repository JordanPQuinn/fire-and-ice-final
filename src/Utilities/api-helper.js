export const fetchApi = async (url) => {
  const initialFetch = await fetch(url)
  const response = await initialFetch.json();
  return response;
}