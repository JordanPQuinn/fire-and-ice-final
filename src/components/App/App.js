import React, { Component } from 'react';
import PropTypes, { shape, func, string } from 'prop-types';
import logo from './logo.svg';
import wolf from './wolf.gif';
import './App.css';
import { fetchApi } from '../../Utilities/api-helper'
import { connect } from 'react-redux';
import { storeHouses } from '../../actions';
class App extends Component {
  constructor(props) {
    super(props)

    this.state = {
      loaded: false
    }

  }

  async componentDidMount() {
    const houseData = await fetchApi('http://localhost:3001/api/v1/houses')
    this.props.storeHouses(houseData)
    setTimeout( () => {
      this.setState({
      loaded: true
    })
    }, 2000)
  }

  render() {
    return (
      <div className='App'>
        <div className='App-header'>
          <img src={logo} className='App-logo' alt='logo' />
          <h2>Welcome to Westeros</h2>
        </div>
        { !this.state.loaded && 
          <img src={wolf} id='wolf' alt='loading-gif' />
        }
        <div className='Display-info'>
        </div>
      </div>
    );
  }
}

// App.propTypes = {
//   fake: shape({ fake: string }),
//   fakeAction: func.isRequired
// };

const mapStateToProps = ({houses}) => ({
  houses
});

const mapDispatchToProps = dispatch => ({ 
  storeHouses: houses => dispatch(storeHouses(houses))
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
