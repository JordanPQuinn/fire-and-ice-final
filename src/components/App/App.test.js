import React from 'react';
import { App } from './App';
import { shallow } from 'enzyme';
import * as api from '../../Utilities/api-helper';
import { mockData } from '../../Utilities/mockData'
import { mapDispatchToProps } from './App';

describe('App Component', () => {
  beforeAll(() => {
    window.fetch = jest.fn().mockImplementation(() => {
      return Promise.resolve({
        ok: true,
        status: 200,
        json: () => Promise.resolve({
          houses: mockData 
        })
      })
    })
  })

  it('should exist', () => {
    const wrapper = shallow(<App storeHouses={jest.fn()}/>);

    expect(wrapper).toMatchSnapshot();
  })

  it('should call fetch on CDM', () => {
    api.getHouses = (url) => {mockData}
    api.dataCleaner = (houses) => {mockData}
    const url = 'http://localhost:3001/api/v1/houses';
    const wrapper = shallow(<App storeHouses={jest.fn()} />)

    expect(window.fetch).toHaveBeenCalledWith(url);
  });

  it('should call dispatch in MDTP', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch());

    expect(mockDispatch).toHaveBeenCalled();
  })

  it('should match the snapshot if the loaded state changes', () => {
    const wrapper = shallow(<App storeHouses={jest.fn()}/>)
    wrapper.setState({loaded:true})

    expect(wrapper).toMatchSnapshot();
  });
});