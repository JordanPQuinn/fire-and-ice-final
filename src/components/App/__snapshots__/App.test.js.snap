// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should exist 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    storeHouses={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="App-header"
>
          <img
                    alt="logo"
                    className="App-logo"
                    src="logo.svg"
          />
          <h2>
                    Welcome to Westeros
          </h2>
</div>,
        <div
          className="Display-info"
>
          <img
                    alt="loading-gif"
                    id="wolf"
                    src="wolf.gif"
          />
</div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="logo"
              className="App-logo"
              src="logo.svg"
/>,
            <h2>
              Welcome to Westeros
</h2>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "App-logo",
              "src": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to Westeros",
            },
            "ref": null,
            "rendered": "Welcome to Westeros",
            "type": "h2",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="loading-gif"
            id="wolf"
            src="wolf.gif"
/>,
          "className": "Display-info",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "loading-gif",
            "id": "wolf",
            "src": "wolf.gif",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="App-header"
>
            <img
                        alt="logo"
                        className="App-logo"
                        src="logo.svg"
            />
            <h2>
                        Welcome to Westeros
            </h2>
</div>,
          <div
            className="Display-info"
>
            <img
                        alt="loading-gif"
                        id="wolf"
                        src="wolf.gif"
            />
</div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className="App-logo"
                src="logo.svg"
/>,
              <h2>
                Welcome to Westeros
</h2>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "App-logo",
                "src": "logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome to Westeros",
              },
              "ref": null,
              "rendered": "Welcome to Westeros",
              "type": "h2",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="loading-gif"
              id="wolf"
              src="wolf.gif"
/>,
            "className": "Display-info",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "loading-gif",
              "id": "wolf",
              "src": "wolf.gif",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`App Component should match the snapshot if the loaded state changes 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    storeHouses={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="App-header"
>
          <img
                    alt="logo"
                    className="App-logo"
                    src="logo.svg"
          />
          <h2>
                    Welcome to Westeros
          </h2>
</div>,
        <div
          className="Display-info"
>
          <Connect(CardContainer) />
</div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="logo"
              className="App-logo"
              src="logo.svg"
/>,
            <h2>
              Welcome to Westeros
</h2>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "App-logo",
              "src": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to Westeros",
            },
            "ref": null,
            "rendered": "Welcome to Westeros",
            "type": "h2",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(CardContainer) />,
          "className": "Display-info",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="App-header"
>
            <img
                        alt="logo"
                        className="App-logo"
                        src="logo.svg"
            />
            <h2>
                        Welcome to Westeros
            </h2>
</div>,
          <div
            className="Display-info"
>
            <Connect(CardContainer) />
</div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className="App-logo"
                src="logo.svg"
/>,
              <h2>
                Welcome to Westeros
</h2>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "App-logo",
                "src": "logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome to Westeros",
              },
              "ref": null,
              "rendered": "Welcome to Westeros",
              "type": "h2",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(CardContainer) />,
            "className": "Display-info",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
